src/python/gudhi/barycenter.py

```diff
@@ -0,0 +1,233 @@
+# This file is part of the Gudhi Library - https://gudhi.inria.fr/ - which is released under MIT.
+# See file LICENSE or go to https://gudhi.inria.fr/licensing/ for full license details.
+# Author(s):       Theo Lacombe
+#
+# Copyright (C) 2019 Inria
+#
+# Modification(s):
+#   - YYYY/MM Author: Description of the modification
+
+
+import ot
+import numpy as np
+import scipy.spatial.distance as sc
+
+from gudhi.wasserstein import _build_dist_matrix, _perstot
+
+
+def _proj_on_diag(w):
+    '''
+        Util function to project a point on the diag.
+    '''
+    return np.array([(w[0] + w[1])/2 , (w[0] + w[1])/2])
+
+
+def _mean(x, m):
+    """
+    :param x: a list of 2D-points, off diagonal, x_0... x_{k-1}
+    :param m: total amount of points taken into account, 
+                that is we have (m-k) copies of diagonal
+    :returns: the weighted mean of x with (m-k) copies of the diagonal
+    """
+    k = len(x)
+    if k > 0:
+        w = np.mean(x, axis=0)
+        w_delta = _proj_on_diag(w)
+        return (k * w + (m-k) * w_delta) / m
+    else:
+        return np.array([0, 0])
+
+
+def _optimal_matching(X, Y, withcost=False):
+    """
+    :param X: numpy.array of size (n x 2)
+    :param Y: numpy.array of size (m x 2)
+    :param withcost: returns also the cost corresponding to the optimal matching
+    :returns: numpy.array of shape (k x 2) encoding the list of edges 
+                                            in the optimal matching. 
+                That is, [[i, j] ...], where (i,j) indicates 
+                that X[i] is matched to Y[j]
+                if i >= len(X) or j >= len(Y), it means they 
+                represent the diagonal.
+                They will be encoded by -1 afterwards.
+    """
+
+    n = len(X)
+    m = len(Y)
+    # Start by handling empty diagrams. Could it be shorten?
+    if X.size == 0: # X is empty
+        if Y.size == 0: # Y is empty
+            res = np.array([[0,0]])  # the diagonal is matched to the diagonal
+            if withcost:
+                return res, 0
+            else:
+                return res
+        else: # X is empty but not Y
+            res = np.array([[0, i] for i in range(m)]) 
+            cost = _perstot(Y, order=2, internal_p=2)**2
+            if withcost:
+                return res, cost
+            else:
+                return res
+    elif Y.size == 0: # X is not empty but Y is empty
+        res = np.array([[i,0] for i in range(n)])        
+        cost = _perstot(X, order=2, internal_p=2)**2
+        if withcost:
+            return res, cost
+        else:
+            return res
+
+    # we know X, Y are not empty diags now
+    M = _build_dist_matrix(X, Y, order=2, internal_p=2)
+
+    a = np.full(n+1, 1. / (n + m) )  
```

Comment:
Did you clarify what happens with integer weights, without renormalization?

Reply:
Well yes, it seems re-normalization is actually useless. I don't know why I thought it was required... (I think that the error message "check that a, b are in the simplex" misled me).
I think it's better to remove it then, it probably reduces numerical error impact.